src/System.Management.Automation/engine/Interop/Windows/WNetAddConnection2.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Windows
+    {
+        internal const int CONNECT_NOPERSIST = 0x00000000;
+        internal const int CONNECT_UPDATE_PROFILE = 0x00000001;
+        internal const int RESOURCE_GLOBALNET = 0x00000002;
+        internal const int RESOURCETYPE_ANY = 0x00000000;
+        internal const int RESOURCEDISPLAYTYPE_GENERIC = 0x00000000;
+        internal const int RESOURCEUSAGE_CONNECTABLE = 0x00000001;
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct NetResource
+        {
+            public int Scope;
+            public int Type;
+            public int DisplayType;
+            public int Usage;
+            public char* LocalName;
+            public char* RemoteName;
+            public char* Comment;
+            public char* Provider;
+        }
+
+        [LibraryImport("mpr.dll", EntryPoint = "WNetAddConnection2W", StringMarshalling = StringMarshalling.Utf16)]
+        internal static partial int WNetAddConnection2(ref NetResource netResource, byte[] password, string userName, int flags);
+
+        internal static unsafe int WNetAddConnection2(string localName, string remoteName, byte[] password, string userName, int connectType)
+        {
+            if (s_WNetApiNotAvailable)
+            {
+                return ERROR_NO_NETWORK;
+            }
+
+            int errorCode = ERROR_NO_NETWORK;
+
+            fixed (char* pinnedLocalName = localName)
+            fixed (char* pinnedRemoteName = remoteName)
+            {
+                NetResource resource = new NetResource()
+                {
+                    Comment = null,
+                    DisplayType = RESOURCEDISPLAYTYPE_GENERIC,
+                    LocalName = pinnedLocalName,
+                    Provider = null,
+                    RemoteName = pinnedRemoteName,
+                    Scope = RESOURCE_GLOBALNET,
+                    Type = RESOURCETYPE_ANY,
+                    Usage = RESOURCEUSAGE_CONNECTABLE
+                };
+
+                try
+                {
+                    errorCode = WNetAddConnection2(ref resource, password, userName, connectType);
+                }
+                catch (System.DllNotFoundException)
+                {
+                    s_WNetApiNotAvailable = true;
```

Comment:
Considering this has been available since Server 2000, is it necessary?

Reply:
Comment in original code says the API absent in Core.

Reply:
Interesting, will have to spin up a Core VM and see if that's true, I can't find any mention that `mpr.dll` is missing from Core.